### PR TITLE
Alfoa/fix win tests

### LINKED ISSUE #982: Fix some Win test
--------
Issue Description
--------
##### What did you expect to see happen?
The new installation of Win caused some tests to fail (in local machines).
This should be addressed

##### What did you see instead?
The new installation procedure will be setup in the regression machine as well so the fixes
will be tested there

##### Do you have a suggested fix for the development team?
explicitly  call bash.exe and python in case of Win

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #982

##### What are the significant changes in functionality due to this change request?
Fixed some tests

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/CodeInterfaces/RELAP7/RELAP7Interface.py [hunk 1]
```diff
@@ -52,10 +52,16 @@ def generateCommand(self,inputFiles,executable,clargs=None,fargs=None,preExec=No
...
+
     if clargs:
       precommand = executable + clargs['text']
     else:
       precommand = executable
+
+    if executable.strip().endswith("py"):
+      # for testing
+      precommand = "python "+ precommand
+
```
Comment (mandd) [inline framework/CodeInterfaces/RELAP7/RELAP7Interface.py]: Why do we need this for the RELAP7 interface?
Should we have it also for other MOOSE based applications?

File: framework/utils/utils.py [hunk 2]
```diff
@@ -1109,3 +1109,40 @@ def getAllSubclasses(cls):
     @ Out, getAllSubclasses, list of class objects for each subclass of cls.
   """
   return cls.__subclasses__() + [g for s in cls.__subclasses__() for g in getAllSubclasses(s)]
+
+def which(cmd):
+  """
+    Emulate the which method in shutil
```
Comment (mandd) [inline framework/utils/utils.py]: Can you expand the description of this function?

File: framework/utils/utils.py [hunk 3]
```diff
@@ -1109,3 +1109,40 @@ def getAllSubclasses(cls):
...
+    return (os.path.exists(fn) and os.access(fn, os.X_OK) and not os.path.isdir(fn))
+  if os.path.dirname(cmd):
+    if _access_check(cmd):
+      return cmd
+    return None
+  path = os.environ.get("PATH", os.defpath)
+  if not path:
+    return None
+  path = path.split(os.pathsep)
+  if sys.platform == "win32":
```
Comment (mandd) [inline framework/utils/utils.py]: basically this "which" function was added to cover the win32 case, right?

File: framework/utils/utils.py [hunk 4]
```diff
@@ -1109,3 +1109,40 @@ def getAllSubclasses(cls):
...
   """
   return cls.__subclasses__() + [g for s in cls.__subclasses__() for g in getAllSubclasses(s)]
+
+def which(cmd):
+  """
+    Emulate the which method in shutil
+    @ In, cmd, str, the exe to check
+    @ Out, which, str, the full path or None if not found
+  """
+  def _access_check(fn):
```
Comment (mandd) [inline framework/utils/utils.py]: should we add a description of this function?

File: framework/CodeInterfaces/RELAP7/RELAP7Interface.py (hunk 1 above)
Comment (alfoa) [reply framework/CodeInterfaces/RELAP7/RELAP7Interface.py]: This is only for testing the dynamic event tree....planning to remove the interface in the future

File: framework/utils/utils.py (hunk 3 above)
Comment (alfoa) [reply framework/utils/utils.py]: basically

File: framework/utils/utils.py (hunk 2 above)
Comment (alfoa) [reply framework/utils/utils.py]: done

File: framework/utils/utils.py (hunk 4 above)
Comment (alfoa) [reply framework/utils/utils.py]: added

**Comment by PaulTalbot-INL** [comment]
Checklists pass, waiting on remainder of automated tests. This has my approval to merge at its current state if the tests pass.

**Comment by PaulTalbot-INL** [comment]
@aalfonsi @alfoa merged.

**Pull request merged**